azurerm/resource_arm_iothub.go

```diff
@@ -907,3 +941,39 @@ func validateIoTHubFileNameFormat(v interface{}, k string) (warnings []string, e
 
 	return warnings, errors
 }
+func expandIPFilterRules(d *schema.ResourceData) *[]devices.IPFilterRule {
+	ipFilterRuleList := d.Get("ip_filter_rule").([]interface{})
```

Comment:
if we make this a Set then we'll want this to be:

```suggestion
	ipFilterRuleList := d.Get("ip_filter_rule").(*schema.Set).List()
```